### ISSUE TITLE
[Bug]: `SimpleSpanProcessor` might fail and error is ignored

### ISSUE THREAD

**Issue by dotansimha** [open-telemetry/opentelemetry-rust#1478]
### What happened?

The `SimpleSpanProcessor` might fail silently on some runtime, for example, when `thread::Builder::new()` is called:

https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/trace/span_processor.rs#L106-L109

The error is ignored, and when Spans are reported, the `self.message_sender.send` call fails with a `SendError` (internally, the error is `Disconnected`).

I was trying to run this on a `wasm32-unknown-unknown` runtime, and it failed silently. Using the global error handler, I was able to get the `Disconnected` error, and then I figured out that it happens where there are not subscribers listening. 

The 

### API Version

0.x

### SDK Version

`opentelemetry_sdk@0.21.2`

### What Exporters are you seeing the problem on?

N/A

### Relevant log output

```shell
failed to spawn exporter thread: Error { kind: Unsupported, message: "operation not supported on
  this platform" }
```

**Comment by cijothomas** [comment]
Would switching SimpleSpanProcessor to be similar to [SimpleLogRecordProcessor](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/src/logs/log_processor.rs#L62) be sufficient? 

This was discussed here : https://github.com/open-telemetry/opentelemetry-rust/issues/1409

**Comment by cijothomas** [comment]
Closing as SimpleSpanProcessor no longer creates own thread.
